### PR TITLE
Merge sdk and catalog docs into kpt

### PR DESCRIPTION
**Pull request by prachirp**
Combine all documentation from kpt-functions-sdk and kpt-functions-catalog with existing kpt docs. Main changes:

- Reorder homepage to move install section to top, add search bar
- Add from kpt-functions-sdk: concepts/functions, guides/consumer/function/pipeline, guides/producer/functions/ts, guides/producer/functions/ts/develop, layout, and static/images
- Add from kpt-functions-catalog: guides/consumer/function/catalog
- Create next steps section at the bottom of all kpt function docs
- Write influences section: concepts/architecture
- Write declarative call section: guides/consumer/function
- Write structured results section: reference/fn/run

Fixes GoogleContainerTools/kpt#640

### CODE REVIEW

File: site/content/en/guides/consumer/function/catalog/_index.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+title: "Functions Catalog"
+linkTitle: "Functions Catalog"
+weight: 8
+type: docs
+description: >
+    Catalog of kpt functions.
+---
+
+<!---
+DO NOT EDIT. Generated by: "cd catalog; npm run gen-docs"
```
Comment (frankfarzan) [inline site/content/en/guides/consumer/function/catalog/_index.md:11]: Where's the package after this move? At very least fully specify the path, but ideally this should be part of a top-level command when generating all docs.

File: site/content/en/guides/producer/functions/ts/_index.md [hunk 2]
```diff
@@ -4,8 +4,103 @@ linkTitle: "TypeScript Function SDK"
...
+   npm run watch
+   ```
+
+1. Run the `label_namespace` function:
+
+   ```sh
+   export CONFIGS=../../example-configs
+
+   kpt fn source $CONFIGS |
+   node dist/label_namespace_run.js -d label_name=color -d label_value=orange |
```
Comment (frankfarzan) [inline site/content/en/guides/producer/functions/ts/_index.md:78]: we should show exec method of running with node.

File: site/content/en/guides/producer/functions/ts/_index.md [hunk 3]
```diff
@@ -4,8 +4,103 @@ linkTitle: "TypeScript Function SDK"
...
+   npm install
+   ```
+
+1. Run the following in a separate terminal to continuously build your function as you make changes:
+
+   ```sh
+   npm run watch
+   ```
+
+1. Run the `label_namespace` function:
```
Comment (frankfarzan) [inline site/content/en/guides/producer/functions/ts/_index.md:72]: In this file we should include the function inline (I think there's a way to link instead of copy code). This gives the reader an idea of what TS functions look like and is consistent with other languages.

File: site/content/en/guides/producer/functions/ts/_index.md [hunk 4]
```diff
@@ -4,8 +4,103 @@ linkTitle: "TypeScript Function SDK"
...
+   in the `example-configs` directory:
+
+   ```sh
+   git diff $CONFIGS
+   ```
+
+1. Try modifying the function in `src/label_namespace.ts` to perform other operations
+   on `example-configs`, then repeat step 4.
+
+   The function should implement the `KptFunc` interface [documented here][api-kptfunc].
```
Comment (frankfarzan) [inline site/content/en/guides/producer/functions/ts/_index.md:92]: Similarly, we can include other interesting functions.

File: site/content/en/guides/producer/functions/ts/develop/_index.md [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+[catalog]: ../../../../consumer/function/catalog
+[api-kptfunc]: https://googlecontainertools.github.io/kpt-functions-sdk/api/interfaces/_types_.kptfunc.html
+[vscode]: https://code.visualstudio.com/
+[npm-packages]: https://github.com/GoogleContainerTools/kpt-functions-sdk/packages
+[download-node]: https://nodejs.org/en/download/
+[download-kind]: https://github.com/kubernetes-sigs/kind
+[install-node]: https://github.com/nodejs/help/wiki/Installation
+[install-docker]: https://docs.docker.com/v17.09/engine/installation
+[beta-feature]: https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.15.md#customresourcedefinition-openapi-publishing
+[run functions]: ../../../../consumer/function/
```
Comment (frankfarzan) [inline site/content/en/guides/producer/functions/ts/develop/_index.md:252]: Can you double check there are no broken links (Changes are high with such moves). Maybe also look into whether there's an automated way to catch broken links.

File: site/content/en/guides/consumer/function/catalog/_index.md (hunk 1 above)
Comment (prachirp) [reply site/content/en/guides/consumer/function/catalog/_index.md:11]: The catalog package is at googlecontainertools.github.io/kpt/guides/consumer/function/catalog. I also included a list of where sdk docs ended up in the description.

Comment (prachirp) [reply site/content/en/guides/consumer/function/catalog/_index.md:11]: Done

File: site/content/en/guides/producer/functions/ts/_index.md (hunk 3 above)
Comment (prachirp) [reply site/content/en/guides/producer/functions/ts/_index.md]: Linked to the function.

File: site/content/en/guides/producer/functions/ts/_index.md (hunk 2 above)
Comment (prachirp) [reply site/content/en/guides/producer/functions/ts/_index.md:78]: What do you mean by this?

File: site/content/en/guides/producer/functions/ts/_index.md (hunk 4 above)
Comment (prachirp) [reply site/content/en/guides/producer/functions/ts/_index.md:92]: Added suggest psp and validate rolebinding as examples of other interesting functions.

File: site/content/en/guides/producer/functions/ts/develop/_index.md (hunk 5 above)
Comment (prachirp) [reply site/content/en/guides/producer/functions/ts/develop/_index.md:248]: I installed and used muffet locally. https://github.com/raviqqe/muffet
I also ran the W3 link checker on the existing kpt, kpt-functions-sdk, and kpt-functions-catalog websites and fixed some of the links it caught.

**Comment by prachirp** [comment]
cc @phanimarupaka Can you take a look at these changes too? Mostly need review of pages unrelated to functions:
- site/content/en/_index.md
- site/content/en/concepts/architecture/_index.md

File: site/content/en/guides/producer/functions/ts/_index.md (hunk 2 above)
Comment (prachirp) [reply site/content/en/guides/producer/functions/ts/_index.md:78]: @frankfarzan I filed https://github.com/GoogleContainerTools/kpt/issues/768 since I wasn't able to get the exec method working. Let's fix this in a separate PR.

File: site/content/en/_index.md [hunk 6]
```diff
@@ -16,6 +16,17 @@ apply Kubernetes configuration.
 
 {{% /blocks/lead %}}
 
+{{< blocks/section color="light" >}}
```
Comment (phanimarupaka) [inline site/content/en/_index.md:19]: What is the motivation behind this change? Can you attach a screenshot as it how it looks like after the change? Do you think the installation icon is not at right place?

Comment (prachirp) [reply site/content/en/_index.md:19]: Having the installation link clearly at the top makes it easier for newcomers to take the first step and install kpt. It also brings us more in line with other websites like docs.docker.com, nodejs.org, helm.sh which all have installation directions first, then guides/reference. Also, the install icon was broken and I fixed it.
Before: 
Screen Shot 2020-06-29 at 11 13 34 AM
After:
Screen Shot 2020-06-29 at 11 14 30 AM

**Comment by prachirp** [comment]
@phanimarupaka Let's merge this. We can address other comments to the site's look-and-feel in followup PRs.

**Comment by frankfarzan** [comment]
Some follow up comments:
- Auto-generated API(s) documetnation (typedoc) is broken now
- Separate Quick Guide into a separate page
- Hugo should have a way to include source functions directly
- We deleted the "how to run functions" page from SDK, was there anything we wanted to carry over?

**Comment by prachirp** [comment]
> Some follow up comments:
> 
> * Auto-generated API(s) documetnation (typedoc) is broken now
Fixing typedoc.
> * Separate Quick Guide into a separate page
Separating Quickstart.
> * Hugo should have a way to include source functions directly
Hugo allows shortcodes so I added one to include source code.
> * We deleted the "how to run functions" page from SDK, was there anything we wanted to carry over?
No, I split the page sections between package publisher and consumer functions docs.
